internal/eval/store_eval.go

```diff
@@ -316,3 +317,628 @@ func evalSETEX(args []string, store *dstore.Store) *EvalResponse {
 
 	return evalSET(newArgs, store)
 }
+
+// SETBIT key offset value
+func evalSETBIT(args []string, store *dstore.Store) *EvalResponse {
+	var err error
+
+	if len(args) != 3 {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrWrongArgumentCount("SETBIT"),
+		}
+	}
+
+	key := args[0]
+	offset, err := strconv.ParseInt(args[1], 10, 64)
+	if err != nil {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrIntegerOutOfRange,
+		}
+	}
+
+	value, err := strconv.ParseBool(args[2])
+	if err != nil {
+		return &EvalResponse{
+			Result: nil,
+			Error:  diceerrors.ErrIntegerOutOfRange,
+		}
+	}
+
+	obj := store.Get(key)
+	requiredByteArraySize := offset>>3 + 1
+
+	if obj == nil {
+		obj = store.NewObj(NewByteArray(int(requiredByteArraySize)), -1, object.ObjTypeByteArray, object.ObjEncodingByteArray)
+		store.Put(args[0], obj)
+	}
+
+	if object.AssertType(obj.TypeEncoding, object.ObjTypeByteArray) == nil ||
+		object.AssertType(obj.TypeEncoding, object.ObjTypeString) == nil ||
+		object.AssertType(obj.TypeEncoding, object.ObjTypeInt) == nil {
+		var byteArray *ByteArray
+		oType, oEnc := object.ExtractTypeEncoding(obj)
+
+		switch oType {
+		case object.ObjTypeByteArray:
+			byteArray = obj.Value.(*ByteArray)
+		case object.ObjTypeString, object.ObjTypeInt:
+			byteArray, err = NewByteArrayFromObj(obj)
+			if err != nil {
+				return &EvalResponse{
+					Result: nil,
+					Error:  diceerrors.ErrWrongTypeOperation,
+				}
+			}
+		default:
+			return &EvalResponse{
+				Result: nil,
+				Error:  diceerrors.ErrWrongTypeOperation,
+			}
+		}
+
+		// Perform the resizing check
+		byteArrayLength := byteArray.Length
+
+		// check whether resize required or not
+		if requiredByteArraySize > byteArrayLength {
+			// resize as per the offset
+			byteArray = byteArray.IncreaseSize(int(requiredByteArraySize))
+		}
+
+		resp := byteArray.GetBit(int(offset))
+		byteArray.SetBit(int(offset), value)
+
+		// We are returning newObject here so it is thread-safe
+		// Old will be removed by GC
+		newObj, err := ByteSliceToObj(store, obj, byteArray.data, oType, oEnc)
+		if err != nil {
+			return &EvalResponse{
+				Result: nil,
+				Error:  diceerrors.ErrWrongTypeOperation,
+			}
+		}
+
+		exp, ok := dstore.GetExpiry(obj, store)
+		var exDurationMs int64 = -1
+		if ok {
+			exDurationMs = int64(exp - uint64(utils.GetCurrentTime().UnixMilli()))
+		}
+		// newObj has bydefault expiry time -1 , we need to set it
+		if exDurationMs > 0 {
+			store.SetExpiry(newObj, exDurationMs)
+		}
+
+		store.Put(key, newObj)
+		if resp {
+			return &EvalResponse{
+				Result: 1,
```

Comment:
Please use `clientio.IntegerOne`

Reply:
For some reason if I use `clientio.IntegerOne` and `clientio.IntegerZero` here the following tests fail: TestGetDel/Getdel_with_value_created_from_Setbit
TestType/TYPE_for_key_with_value_created_from_SETBIT_command.

I am not sure why.
Any suggestions?

Reply:
This is failing in http tests. 
Please change the output of these tests in the following files

1. https://github.com/DiceDB/dice/blob/master/integration_tests/commands/http/type_test.go#L71 : float64(0) -> "0"
2. https://github.com/DiceDB/dice/blob/master/integration_tests/commands/http/getdel_test.go#L67 : float64(0) -> "0"